src/Nancy/NancyEngine.cs

```diff
@@ -217,17 +217,42 @@ private void CheckStatusCodeHandler(NancyContext context)
                 return;
             }
 
-            var handlers = this.statusCodeHandlers
-                .Where(x => x.HandlesStatusCode(context.Response.StatusCode, context))
-                .ToList();
+            IStatusCodeHandler defaultHandler = null;
+            IStatusCodeHandler customHandler = null;
 
-            var defaultHandler = handlers
-                .FirstOrDefault(x => x is DefaultStatusCodeHandler);
+            foreach (var statusCodeHandler in this.statusCodeHandlers)
+            {
+                if (statusCodeHandler.HandlesStatusCode(context.Response.StatusCode, context))
+                {
+                    if (defaultHandler == null && (statusCodeHandler is DefaultStatusCodeHandler))
+                    {
+                        defaultHandler = statusCodeHandler;
+                    }
+
+                    if (customHandler == null && !(statusCodeHandler is DefaultStatusCodeHandler))
+                    {
+                        customHandler = statusCodeHandler;

```

Comment:
same here. it's not possible that a handler can be both custom and default, right?